### PR TITLE
Gdcc/8746 single version semantics for archiving

### LINKED ISSUE #8746: HDC 3A: Support single-version-only archiving
**Overview of the Feature Request** Some archival systems, including the current Harvard DRS system, only support archiving a single version of a dataset. The Harvard Data Commons 3A Objective has  a requirement to support this - allowing archivers to indicate that the only support a single version and for the archiving mechanism, status mechanism and APIs, UI etc. to then support that. While conceptually stand-alone, the accompanying PR builds on #8696 so it can add the changes to the new status API.

### PR DESCRIPTION
**Pull request by qqmyers**
**What this PR does / why we need it**: This PR adds a mechanism for archivers to specify that they only allow archiving for a single version and implements logic to enforce that in the archiving/archival status APIs

**Which issue(s) this PR closes**:

Closes #8746

**Special notes for your reviewer**: Review after #8696 as this incorporates the status API updates from that PR. Hopefully that PR can be merged and this will be updated before review.

**Suggestions on how to test this**: The DRS Archiver (not yet in a PR) is the only archiver that supports single-version mode at present, so testing of this by itself could probably be limited to assuring that the Local archiver still works/still allows you to archive multiple versions.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**: Addressed in #8611

**Additional documentation**:

### CODE REVIEW

File: doc/sphinx-guides/source/api/native-api.rst [hunk 1]
```diff
@@ -1873,6 +1873,63 @@ The API call requires a Json body that includes the list of the fileIds that the
   export JSON='{"fileIds":[300,301]}'
 
   curl -H "X-Dataverse-key: $API_TOKEN" -H "Content-Type:application/json" "$SERVER_URL/api/datasets/:persistentId/files/actions/:unset-embargo?persistentId=$PERSISTENT_IDENTIFIER" -d "$JSON"
+  
+  
+Get the Archival Status of a Dataset By Version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Archiving is an optional feature that may be configured for a Dataverse instance. When enabled, this API call be used to retrieve the status. Note that this requires "superuser" credentials.
```
Comment (sekmiller) [inline doc/sphinx-guides/source/api/native-api.rst:1881]: Kind.of a quibble, but are we using "installation" as the term here or instance?

Comment (qqmyers) [reply doc/sphinx-guides/source/api/native-api.rst]: FWIW: A quick scan through the guides shows some of each - installation is more popular. I see a dozen+ 'Dataverse instance' and lots of 'instance' referring to other components.

**Comment by qqmyers** [comment]
@sekmiller - I've updated this from #8696 so it has all the changes from that review. Which means comparing against that branch should show just the single version code.

File: doc/sphinx-guides/source/api/native-api.rst (hunk 1 above)
Comment (pdurbin) [reply doc/sphinx-guides/source/api/native-api.rst]: FWIW, I prefer "Dataverse installation" to "Dataverse instance".

**Review by sekmiller — APPROVED** [review]

**Pull request merged**